iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -541,80 +605,338 @@ LogicalResult preProcessInterfaceStoreTensorOp(
+    });
+  }
+  return resultBuffer;
+}
+
+/// Converts a `tensor.cast` operation into a `memref.cast` operation with the
+/// result aliasing the buffer for the operand.
+static Value getAliasingBufferForCastResult(OpBuilder &b, tensor::CastOp castOp,
+                                            BlockAndValueMapping &bvm) {
+  Value inputBuffer = bvm.lookup(castOp.source());
+  Value resultTensor = castOp.dest();
+  auto outputType = getMemrefTypeForTensor(
+      resultTensor.getType().cast<RankedTensorType>(), {},
+      inputBuffer.getType().cast<MemRefType>().getMemorySpaceAsInt());
+  return b.create<memref::CastOp>(castOp.getLoc(), outputType, inputBuffer);
+}
+
+/// Converts a `linalg.tensor_reshape` operation to a `linalg.reshape`
+/// operation with the result aliasing the buffer for the operand.
+static Value getAliasingBufferForReshapeResult(OpBuilder &b,
+                                               linalg::TensorReshapeOp op,
+                                               BlockAndValueMapping &bvm) {
+  Location loc = op.getLoc();
+  Value srcTensor = op.src();
+  RankedTensorType resultTensorType = op.getResultType();
+  Value inputBuffer = bvm.lookup(srcTensor);
+
+  // Create the reshape op.
+  MemRefType inputBufferType = inputBuffer.getType().cast<MemRefType>();
+  auto reshapeResultType = getMemrefTypeForTensor(
+      resultTensorType, {}, inputBufferType.getMemorySpaceAsInt());
+  Value bufferReshape = b.create<linalg::ReshapeOp>(
+      loc, reshapeResultType, inputBuffer, op.reassociation());
+  return bufferReshape;
+}
+
+/// Converts a `subtensor` operation to a `subview` operation.
+static Value getAliasingBufferForSubtensorResult(OpBuilder &b, SubTensorOp op,
+                                                 BlockAndValueMapping &bvm) {
+  Location loc = op.getLoc();
+  Value srcTensor = op.source();
+  Value inputBuffer = bvm.lookup(srcTensor);
+
+  ShapedType sourceType = op.getSourceType();
+  ShapedType resultType = op.getType();
+  SmallVector<OpFoldResult> offsets = op.getMixedOffsets();
+  SmallVector<OpFoldResult> sizes = op.getMixedSizes();
+  SmallVector<OpFoldResult> strides = op.getMixedStrides();
+  MemRefType subViewResultType =
+      (resultType.getRank() < sourceType.getRank()
+           ? memref::SubViewOp::inferRankReducedResultType(
+                 resultType.getRank(), inputBuffer.getType().cast<MemRefType>(),
+                 offsets, sizes, strides)
+                 .cast<MemRefType>()
+           : MemRefType());
+  return b.create<memref::SubViewOp>(loc, subViewResultType, inputBuffer,
+                                     offsets, sizes, strides);
+}
+
+/// Computes the `memrefs` to use for the result of an operation based on
+/// - if the result has a tied operand reuse the buffer for the tied operand (or
+///   an alias of it) as the buffer for the result. The `tiedOperands` vector is
+///   expected to be as large as the number of results. If the result has no
```

Comment:
Should this if be put into another item?
```suggestion
///   expected to be as large as the number of results.
/// - If the result has no ...
```